modules/yardstick/src/main/java/org/apache/ignite/yardstick/cache/load/IgniteCacheRandomOperationBenchmark.java

```diff
@@ -713,229 +846,283 @@ private void doSqlQuery(IgniteCache cache) throws Exception {
+     * @param val Value.
+     */
+    ContinuousQueryFilter(int val) {
+        this.val = val;
+    }
 
-            for (Integer part : myPartitions) {
-                if (ThreadLocalRandom.current().nextBoolean())
-                    continue;
+    /** {@inheritDoc} */
+    @Override public boolean evaluate(CacheEntryEvent evt) throws CacheEntryListenerException {
+        return evt.getOldValue() != null && evt.getValue() != null
+            && ((evt.getOldValue().hashCode() - evt.getValue().hashCode()) % (val) == 0);
+    }
+}
 
-                ScanQuery scanQry = new ScanQuery();
+/**
+ * Closure for scan query executing.
+ */
+private static class ScanQueryBroadcastClosure implements IgniteRunnable {
+    /**
+     * Ignite node.
+     */
+    @IgniteInstanceResource
+    private Ignite node;
 
-                scanQry.setPartition(part);
-                scanQry.setFilter(igniteBiPred);
+    /**
+     * Information about partition.
+     */
+    private Map<UUID, List<Integer>> cachePart;
 
-                try (QueryCursor cursor = cache.query(scanQry)) {
-                    for (Object obj : cursor)
-                        ;
-                }
+    /**
+     * Name of Ignite cache.
+     */
+    private String cacheName;
+
+    /**
+     * @param cacheName Name of Ignite cache.
+     * @param cachePart Partition by node for Ignite cache.
+     */
+    private ScanQueryBroadcastClosure(String cacheName, Map<UUID, List<Integer>> cachePart) {
+        this.cachePart = cachePart;
+        this.cacheName = cacheName;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void run() {
+        IgniteCache cache = node.cache(cacheName);
+
+        // Getting a list of the partitions owned by this node.
+        List<Integer> myPartitions = cachePart.get(node.cluster().localNode().id());
+
+        for (Integer part : myPartitions) {
+            if (ThreadLocalRandom.current().nextBoolean())
+                continue;
 
+            ScanQuery scanQry = new ScanQuery();
+
+            scanQry.setPartition(part);
+            scanQry.setFilter(igniteBiPred);
+
+            try (QueryCursor cursor = cache.query(scanQry)) {
+                for (Object obj : cursor)
+                    ;
             }
+
         }
     }
+}
+
+/**
+ * Scan query predicate class.
+ */
+private static class BenchmarkIgniteBiPredicate implements IgniteBiPredicate {
 
     /**
-     * Scan query predicate class.
+     * @param key Cache key.
+     * @param val Cache value.
+     * @return true If is hit.
      */
-    private static class BenchmarkIgniteBiPredicate implements IgniteBiPredicate {
-
-        /**
-         * @param key Cache key.
-         * @param val Cache value.
-         * @return true If is hit.
-         */
-        @Override public boolean apply(Object key, Object val) {
-            return val.hashCode() % 45 == 0;
-        }
+    @Override public boolean apply(Object key, Object val) {
+        return val.hashCode() % 45 == 0;
```

Comment:
Why 45? Actually there is no need to use the filter. Let's test the scenario when all entries stored in partition has to be analyzed by iterator.